qcodes_contrib_drivers/drivers/Spectrum/M4i.py

```diff
@@ -695,59 +733,152 @@ def _read_channel(self, channel, memsize=None):
+                0x01000001: Positive + re-arm, 0x01000002: Negative + re-arm
+            termination: input termination 0: 1 MOhm, 1: 50 Ohm
+            coupling: DC/AC input coupling (0: DC, 1: AC)
+            level0: trigger level [mV]
+            level1: 2nd level for re-arm and windowed modes. [mV]
+        '''
         self.channel_or_mask(0)
-        self.external_trigger_mode(trig_mode)  # trigger mode
-        self.trigger_or_mask(pyspcm.SPC_TMASK_EXT0)  # external trigger
-        self.external_trigger_termination(termination)  # 1: 50 Ohm
-        self.external_trigger_input_coupling(coupling)  # 0: DC
-        self.external_trigger_level_0(level0)  # mV
+        self.external_trigger_mode(trig_mode)
+        self.trigger_or_mask(pyspcm.SPC_TMASK_EXT0)
+        self.external_trigger_termination(termination)
+        self.external_trigger_input_coupling(coupling)
+        self.external_trigger_level_0(level0)
         if(level1 != None):
-            self.external_trigger_level_1(level1)  # mV
+            self.external_trigger_level_1(level1)
 
     # Note: the levels need to be set in bits, not voltages! (between -8191 to
     # 8191 for 14 bits)
     def set_channel_OR_trigger_settings(self, i, trig_mode, bitlevel0, bitlevel1=None):
         """When a channel is used for triggering it must be enabled during the
         acquisition."""
         self.trigger_or_mask(0)
-        self.channel_or_mask(getattr(pyspcm, 'SPC_TMASK0_CH{}'.format(i)))
-        getattr(self, 'trigger_channel_{}_level_0'.format(i))(bitlevel0)
+        self.channel_or_mask(getattr(pyspcm, f'SPC_TMASK0_CH{i}'))
+        self.set(f'trigger_channel_{i}_level_0', bitlevel0)
         if(bitlevel1 != None):
-            getattr(self, 'trigger_channel_{}_level_1'.format(i))(bitlevel1)
-        getattr(self, 'trigger_mode_channel_{}'.format(i))(
-            trig_mode)  # trigger mode
+            self.set(f'trigger_channel_{i}_level_1', bitlevel1)
+        self.set(f'trigger_mode_channel_{i}', trig_mode)  # trigger mode
+
+    def setup_multi_recording(self, posttrigger_size, n_triggers=1,
+                              pretrigger_size=None, boxcar_average=False):
+        """ Setup multi recording.
+
+        Triggering must have been configured separately.
+        Data acquisition must started with start_triggered().
+        The collected data can be acquired with the function get_data().
+
+        Args:
+            posttrigger_size (int): size of data trace after triggering
+            n_triggers (int): total number of triggers
+            pretrigger_size (int): size of data trace before triggering
+            boxcar_average (bool): use mode 'boxcar average'
+
+        Example:
+            digitizer.setup_multi_recording(size, n_triggers)
+            digitizer.start_triggered()
+            data = digitizer.get_data()
+            digitizer.start_triggered()
+            data = digitizer.get_data()
```

Comment:
Not sure why digitizer.start_triggered() and data = digitizer.get_data() are repeated here?